promql/engine.go

```diff
@@ -668,16 +691,19 @@ func (enh *EvalNodeHelper) signatureFunc(on bool, names ...string) func(labels.L
 
 // rangeEval evaluates the given expressions, and then for each step calls
 // the given function with the values computed for each expression at that
-// step.  The return value is the combination into time series of  of all the
+// step.  The return value is the combination into time series of all the
 // function call results.
 func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ...Expr) Matrix {
 	numSteps := int((ev.endTimestamp-ev.startTimestamp)/ev.interval) + 1
 	matrixes := make([]Matrix, len(exprs))
 	origMatrixes := make([]Matrix, len(exprs))
+	tempNumSamples := ev.currentSamples
 	for i, e := range exprs {
 		// Functions will take string arguments from the expressions, not the values.
 		if e != nil && e.Type() != ValueTypeString {
+			tempNumSamples = ev.currentSamples
 			matrixes[i] = ev.eval(e).(Matrix)
+			ev.currentSamples = tempNumSamples + matrixes[i].TotalSamples()
```

Comment:
ev.currentSamples will already have the right value due to the eval, no need to change it